text/0002-new-architecture.md

```diff
@@ -0,0 +1,243 @@
+* Start Date: 2022-07-21
+* RFC Type: informational
+* RFC PR: https://github.com/getsentry/rfcs/pull/2
+
+# Summary
+
+This document is a living piece that is updated with some collected thoughts on what likely projects
+are over a multi year horizon to improve Sentry's underpinnings.
+
+# Motivation
+
+We are running into scaling limitations on our current infrastructure and as such some larger
+decisions have to be made about the future of our codebase.  As we are talking about creating
+more Sentry deployments in different regions and for smaller clusters, some of these problems
+are becoming more pressing as they create complexity for each added installation.
+
+# Expected Workstreams
+
+This captures all expected streams of work that are in the scope of overhauling Sentry's
```

Comment:
Another aspect we may want to capture regarding the new architecture: the concept of ownership and self service. We have seen that we cannot live anymore in a world where ops and SnS own everything and that product teams need to take ownership of their system, which can include application code in the monolith but also systems (like kafka consumers or storage instances).  This also includes being accountable for the resources they use so that they can take responsibility for decisions between a short term non-scalable solution to deliver faster and a long term scalable solution.

I think there are two fundamental ways to approach the problem that generate different architectures:
- physically segregate systems by ownership. Product would have full stack ownership of their resources including k8s deployments, clickhosue clusters, k8s namespace.
- a mixed model where the product owns application logic running on a serverless infrastructure managed by infra teams. 

For snuba we are going towards the second model. I am outlining it here https://www.notion.so/sentry/Improve-product-development-speed-92c5b9f06a6744288454437f5a8c3db6

Though this problem will apply to Sentry as well.

Reply:
@fpacifici would be nice to move this into a public space so I can reference it: https://www.notion.so/sentry/Improve-product-development-speed-on-Snuba-92c5b9f06a6744288454437f5a8c3db6